### ISSUE TITLE
Power Query style Table.Profile over DAX calculated tables [enhancement]

### ISSUE THREAD

**Issue by sivajan** [DaxStudio/DaxStudio#114]
DAX Studio could help users by providing a button to obtain table statistics on a DAX calculated table created to query a tabular model from within DAX Studio. The table statistics could be similar to the Table.Profile function available in Power Query.

**Comment by marcosqlbi** [comment]
Can you provide a use case for that? I understand one would like to retrieve statistics for an existing table, but why on a calculated table? In order to do that you'd need to materialize the entire result and if the table is large it would be extremely expensive. When would you use it?

**Comment by sivajan** [comment]
Generally, this can be useful when users want to create a calculated table in the model but would like to review its statistics, for example to check data quality or to confirm other understood business logic. A more specific example is to split a table that has both dimensions and facts into separate dimension and fact tables using DAX. To perform a quick data quality check that the primary key of the new DAX calculated dimension table is unique, and it references only one value for each of the other dimensional columns. I  understand that as the model grows, these data quality checks would need to be reviewed and enforced at the data source (e.g. DW).

**Comment by sivajan** [comment]
Thanks Marco - yes, that actually is a good workaround given that this would require materialization of the table. Would it be possible for DAX Studio to perform this by only temporarily materializing to obtain the table stats?

**Comment by marcosqlbi** [comment]
No, writing data from DAX Studio in the model is not supported. I'm not saying it's impossible, but DAX Studio should:
1) Create a temporary table (waiting the necessary time)
2) Export the DMV for that table
3) Remove the calculated table
Considering that the time you would save is minimal, especially using Power BI, I don't think that this would worth the effort. But DAX Studio is open source so it's open to other contributions!

**Comment by dgosbell** [comment]
@sivajan I think for your example scenario that the existing Table.Profile in Power Query functionality is much better suited. Any normalizing / normalizing style activities are usually better done as early in your data pipeline as possible.
 
But, I do think that this sort of feature could be of interest and I have been toying with the idea of adding some sort of profiling/analytics. I'm even wondering if this idea of creating and analyzing a temporary table could help with measuring the impact of materialization during query execution. But I'm not sure if that sorts of temporary materialization incurs all the same overheads as creating an explicit calculated table. 

We could do something like:

1. Begin Tran
2. Create a temporary table
3. Run the DMV queries
4. Rollback the transaction

The advantage of using a transaction is that if anything interrupts the process it should just rollback. And in the case of something like SSAS or eventually a PowerBI XMLA endpoint - that the data inside the uncommitted transaction would not be visible in any other sessions.

**Comment by marcosqlbi** [comment]
The entire calculated table must be materialized uncompressed in memory and then compressed. The DMV would return the size of the compressed table, not the size of the materialization, which is already available (as an estimate) in the Storage Engine events. Thus, you would not measure the real cost of the materialization, which is already available in existing metrics.

**Comment by dgosbell** [comment]
> The entire calculated table must be materialized uncompressed in memory and then compressed. The DMV would return the size of the compressed table, not the size of the materialization, which is already available (as an estimate) in the Storage Engine events.

Ah, of course. Given this I'm struggling to come up with a use case for temporarily materializing a table that provides any value beyond what can already be done in other tools like Power Query or Vertipaq Analyzer
